network/transport/v2/conversation.go

```diff
@@ -56,8 +56,9 @@ func (cid conversationID) String() string {
 }
 
 type conversation struct {
-	conversationID   conversationID
-	createdAt        time.Time
+	conversationID conversationID
+	// lastValidAt is the time the conversation was created or when the last message for the conversation was successfully processed.
+	lastValidAt      time.Time
```

Comment:
`lastValidAt` sounds a bit quirky, would it make more sense to call it `expiry`? Then you only need to do calculations with `Add()` on it in `resetTimeout`.